libraries/chain/worker_proposal_evaluators.cpp

```diff
@@ -0,0 +1,61 @@
+#include <golos/protocol/worker_proposal_operations.hpp>
+#include <golos/protocol/exceptions.hpp>
+#include <golos/chain/steem_evaluator.hpp>
+#include <golos/chain/database.hpp>
+#include <golos/chain/steem_objects.hpp>
+#include <golos/chain/worker_proposal_objects.hpp>
+
+namespace golos { namespace chain {
+
+    void worker_proposal_evaluator::do_apply(const worker_proposal_operation& o) {
+        ASSERT_REQ_HF(STEEMIT_HARDFORK_0_20__1013, "worker_proposal_operation");
+
+        const auto& comment = _db.get_comment(o.author, o.permlink);
+
+        GOLOS_CHECK_LOGIC(comment.parent_author == STEEMIT_ROOT_POST_PARENT,
+            logic_exception::worker_proposal_can_be_created_only_on_post,
+            "Worker proposal can be created only on post");
```

Comment:
if we sync cashout window with proposal approve window, there must be check that post is still open